src/lib/y2partitioner/widgets/device_table_entry.rb

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Y2Partitioner
+  module Widgets
+    # Class to represent each entry of a table of devices, including the device
+    # itself and all the corresponding nested entries (like the partitions of a
+    # given disk).
+    #
+    # For example, a table containing two disks with several partitions each would
+    # have two top-level DeviceTableEntry objects, each of them containing the
+    # corresponding partitions as nested ({#children}) DeviceTableEntry objects.
+    class DeviceTableEntry
+      # Device represented in this entry
+      # @return [Y2Storage::Device, Y2Storage::SimpleEtcFstabEntry]
+      attr_accessor :device
+
+      # Nested entries
+      #
+      # @return [Array<DeviceTableEntry>]
+      attr_reader :children
+
+      # Whether the device should be represented in the table using the full name
+      #
+      # When this is false, the device may be represented with the full name or
+      # with a short version, depending on the circumstances.
+      #
+      # @return [Boolean] true to enforce the usage of full name
+      def full_name?
```

Comment:
NP: this method is intended to "force" a full name, but not to decide whether to use a short or full name. I would use a more specific name to avoid misleading, e.g., `#force_full_name?`.

Reply:
That was my first intention, but I found it to be too long